### ISSUE TITLE
consider a different promise library?

### ISSUE THREAD

**Issue by lalitkapoor** [vitaly-t/pg-promise#4]
Hi, would you consider using the bluebird promise library for performance reasons? Thanks.

**Comment by vitaly-t** [comment]
Hi,

There is one already: https://github.com/Aphel-Cloud-Solutions/pg-bluebird

It is a far more simplistic one, doesn't really do much, if you look at the code. And I argued with the author that even his one code example is incorrect, the way he closes the database connection.

I believe, that with a little tweak I could provide a switch from one promise library to another, provided there is a 100% compatibility, which I'm not sure even when it comes to between Promise and Bluebird.

For example, I'm using method done() to release a connection, not sure Bluebird has one, because it is not quite from A+ specification.

Anyhow, to sum this up, yes, I think this is quite doable, just takes a little time and effort ;)

Just so, when it comes to making calls to the database, performance of the promise library becomes absolutely irrelevant, as it will always be less than 1% of time from the query execution.

So, just for the performance reasons, the way you phrased it - I wouldn't consider it, only for the reason of flexibility of which promise library to use.

**Comment by lalitkapoor** [comment]
What does the done method have to do with the promise library? That's just the function that's returned when opening a connection. Sorry, I don't quite follow.

**Comment by vitaly-t** [comment]
I referred to method done() of the Promise library, not one from the PG library :)

**Comment by lalitkapoor** [comment]
Oh, I see. Bluebird does have a done as well https://github.com/petkaantonov/bluebird/blob/master/API.md#donefunction-fulfilledhandler--function-rejectedhandler----void

**Comment by lalitkapoor** [comment]
Also, you can see more about speed and memory usage across various implementaitons and how they compare here - https://github.com/petkaantonov/bluebird/tree/master/benchmark

**Comment by vitaly-t** [comment]
Wonderful, then we are almost done here :)

This means all my code samples will work just the same, if you make a request from Promise and then continue the resolution chain through Bluebird.

Surely, someone must have tried it already. 

And as I stated earlier, performance of promises isn't relevant when it comes to executing database queries.

**Comment by vitaly-t** [comment]
You know, I'm beginning to like your idea. I've just checked in an initial support for the feature, new property `promiseLib' is now supported within the library options:
```javascript
var promise = require('bluebird');
var options = {
    promiseLib: promise
};
var pgp = pgpLib();
```

I've tested it against the whole library, and everything seems to work just fine. That was easy :)

I'm gonna give a bit more testing and will include it into the next release ;)

Cheers!
